core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -632,9 +633,11 @@ private void retry(Workflow workflow) {
             }
         }
 
-        if (retriableMap.values().size() == 0) {
+        // if workflow TIMED_OUT due to timeoutSeconds configured in the workflow definition,
+        // it may not have any unsuccessful tasks that can be retried
+        if (retriableMap.values().size() == 0 && workflow.getStatus() != WorkflowStatus.TIMED_OUT) {
```

Comment:
Is it to account for the scenario, when the last task was successful and the workflow is TIMED_OUT before scheduling the next task?

Reply:
yes, because the lazy timeout evaluation will mark the workflow as TIMED_OUT before scheduling the next task.